diff --git a/arch/riscv/boot/dts/allwinner/sun20i-d1-nezha.dts b/arch/riscv/boot/dts/allwinner/sun20i-d1-nezha.dts
index 4ed33c1e7c9c..ff387fb1dc2c 100644
--- a/arch/riscv/boot/dts/allwinner/sun20i-d1-nezha.dts
+++ b/arch/riscv/boot/dts/allwinner/sun20i-d1-nezha.dts
@@ -87,10 +87,10 @@ &ehci1 {
 };
 
 &emac {
-       pinctrl-0 = <&rgmii_pe_pins>;
+       pinctrl-0 = <&rmii_pe_pins>;
        pinctrl-names = "default";
        phy-handle = <&ext_rgmii_phy>;
-       phy-mode = "rgmii-id";
+       phy-mode = "rmii";
        phy-supply = <&reg_vcc_3v3>;
        status = "okay";
 };
@@ -124,6 +124,7 @@ pcf8574a: gpio@38 {
 &mdio {
        ext_rgmii_phy: ethernet-phy@1 {
                compatible = "ethernet-phy-ieee802.3-c22";
+               smsc,disable-energy-detect;
                reg = <1>;
        };
 };


